packages/roosterjs-editor-plugins/lib/plugins/TableSelection/TableSelection.ts

```diff
@@ -0,0 +1,195 @@
+import { EditorPlugin, IEditor, Keys, PluginEvent, PluginEventType } from 'roosterjs-editor-types';
+import {
+    Browser,
+    clearSelectedTableCells,
+    highlightTableSelection,
+    setTableSelectedRange,
+} from 'roosterjs-editor-dom';
+
+/**
+ * TableSelectionPlugin help highlight table cells
+ */
+export default class TableSelectionPlugin implements EditorPlugin {
+    private editor: IEditor;
+    private mouseUpEventListerAdded: boolean;
+    private lastTarget: EventTarget;
+    private firstTarget: EventTarget;
+
+    /**
+     * Get a friendly name of  this plugin
+     */
+    getName() {
+        return 'TableSelection';
+    }
+
+    /**
+     * Initialize this plugin. This should only be called from Editor
+     * @param editor Editor instance
+     */
+    initialize(editor: IEditor) {
+        this.editor = editor;
+        console.log(this.editor.getScrollContainer());
+        setTimeout(() => {
+            const styleElement = document.createElement('style');
+            styleElement.id = 'Style4TableSelection';
+            this.editor.getDocument().head.append(styleElement);
+
+            let sheet = styleElement.sheet;
+            // sheet.insertRule(
+            //     'td.TableCellSelected *::selection { background-color: red !important;}'
+            // );
+            // sheet.insertRule(
+            //     'td.TableCellSelected::selection { background-color: red !important;}'
+            // );
+
+            sheet.insertRule(
```

Comment:
As I said, when select table sell, do not let browser handle selection. So we just need to call event.preventDefault() to disable it. Then we know which cells mouse is over and we use our own data structure to manage selection.

Reply:
On what element should we add the event listener?
I think the event selection change, on the document element it is not cancellable.
Also adding a select event listener on the elements is not cancellable, 
https://developer.mozilla.org/en-US/docs/Web/API/Element/select_event 

I tried adding prevent default on the mouse move event and it is not preventing the selection.

I also tried the selectstart event, but if the user is starting the selection outside of the table it is not working.

Reply:
![table selection](https://user-images.githubusercontent.com/23065085/142464006-2312d966-ce85-4639-b0c6-e0251840a6c8.gif)

I did a little change on your branch:

![image](https://user-images.githubusercontent.com/23065085/142464115-2878c363-a31b-4b26-b5c9-596f9772828c.png)